webhook-service/lib/curl_validator.go

```diff
@@ -0,0 +1,44 @@
+package lib
+
+import (
+	"fmt"
+	"strings"
+)
+
+type CurlValidator struct {
+	denyListProvider IDenyListProvider
+	ipResolver       IIPResolver
+}
+
+type ICurlValidator interface {
```

Comment:
Let's call the interface `RequestValidator` and the implemenataion `requestValidator`